### ISSUE TITLE
WinPython 3.7.0.1 should have astroid 2.0 (or 2.0.1)

### ISSUE THREAD

**Issue by brianarichter** [spyder-ide/spyder#7537]
### What steps will reproduce the problem?

I just installed WinPython64-3.7.0.1 on my Windows 10 machine.
I entered the file below into the Spyder editor and pressed F8, and instead of an analysis, a window popped up with this:

### START OF ERROR TEXT
    Traceback (most recent call last):
      File "C:\WPy-3701\python-3.7.0.amd64\lib\site-packages\astroid\decorators.py", line 89 in wrapped
        res = next(generator)
      StopIteration

    During handling of the above exception, another exception occurred:

    Traceback (most recent call last):
      File "C:\WPy-3701\python-3.7.0.amd64\lib\site-packages\astroid\decorators.py", line 89 in wrapped
        res = next(generator)
      StopIteration

    The above exception was the direct cause of the following exception:
        <...bunch of other stuff leading up to...>
      File "C:\Wpy-3701\python-3.7.0.amd64\lib\site-packages\pylint\checkers\refactoring.py", line 365, in _check_raising_stopiteration_in_generator_next_call
        inferred = utils.safe_infer(node.func)
      File "C:\WPy-3701\python-3.7.0.amd64\lib\site-packages\pylint\checkers\utils.py", line 826, in safe_infer
        next(inferit)
      RuntimeError: generator raised StopIteration
### END OF ERROR TEXT

The problem did not occur with WinPython64-3.6.6.1Qt5.

Looking at the original exception, a TODO comment in astroid indicated that for that version (1.6.5), some work needed to be done to adapt to Python 3.7.

I tried upgrading to the latest astroid (2.0.1) and this FIXED the problem.

The file was called "test1.py" and was in a one-file Spyder project called "project1":

### START OF FILE
# -*- coding: utf-8 -*-
"""
Spyder Editor

This is a temporary script file.
"""

def testfunc(arg1, arg2):
    """A test function to add two numbers."""
    return arg1 + arg2

def testfunc2(arg3, arg4):
    """A test function to multiply two numbers."""
    return arg3 * arg4

def testfunc3():
    """A test function to call stuff."""
    return testfunc("aaa", "bbb")

## Versions

* Spyder version: 3.3.0 
* Python version: 3.7.0
* Qt version: 5.9.3
* PyQt5 version: 5.9.2
* Operating System: Windows 10

### Dependencies

```
pyflakes >=0.6.0  :  2.0.0 (OK)
pycodestyle >=2.3 :  2.4.0 (OK)
pygments >=2.0    :  2.2.0 (OK)
sphinx >=0.6.6    :  1.7.5 (OK)
rope >=0.9.4      :  0.10.7 (OK)
jedi >=0.9.0      :  0.12.1 (OK)
nbconvert >=4.0   :  5.3.1 (OK)
pandas >=0.13.1   :  0.23.3 (OK)
numpy >=1.7       :  1.14.5 (OK)
sympy >=0.7.3     :  1.1.1 (OK)
cython >=0.21     :  0.28.3 (OK)
qtconsole >=4.2.0 :  4.3.1 (OK)
IPython >=4.0     :  6.4.0 (OK)
matplotlib >=2.0.0:  2.2.2 (OK)
pylint >=0.25     :  1.9.2 (OK)
```

**Comment by ccordoba12** [comment]
Please report this in WinPython:

https://github.com/winpython/winpython/

**Comment by stonebig** [comment]
well, WinPython-3.7.0.1 was out before everyone could fixed some compatibility issues, as AppVeyor and Travis-ci didn't have their Python-3.7 stack working.

next WinPython-3.7 beta will have astroid fix, scikit-learn fix, and other python-3.7 fixes. 

If you want something  stable, always wait the 3rd iterative release of anything.
